app/components/Texting/Texting.tsx

```diff
@@ -8,31 +8,28 @@ import { Loader } from "../ui";
 import { SentView } from "./components/SentView";
 import { ErrorView } from "./components/ErrorView";
 
-/** A view of a Service from within the Texting component.
- *
- * TODO: Replace this with an app-wide Service type.
- */
-export interface TextingService {
-  serviceName: string;
-  serviceId: number;
+export interface TextListing {
+  listingName: string;
+  serviceId?: number;
+  resourceId?: number;
 }
```

Comment:
I'm inspired by the DB constraint you added on the backend side, and if you wanted to do something similar here, you could write this as:

```suggestion
export type TextListing = {listingName: string} & ({serviceId: number} | {resourceId: number});
```

Or slightly less fancy:

```suggestion
export type TextListing = {listingName: string; serviceId: number} | {listingName: string; resourceId: number};
```

Reply:
Great suggestion, thanks!

Reply:
Ultimately, I also had to add a "type" prop to the interface as the compiler was complaining when I would try to read a resourceId or serviceId off of a TextListing record, given either prop potentially does not exist. Adding the type and using it as a check satisfied the compiler. Anyhow, great improvement and will prevent possible bugs, thanks!